GitCommands/Git/AheadBehindDataProvider.cs

```diff
@@ -23,20 +25,37 @@ public class AheadBehindDataProvider : IAheadBehindDataProvider
                    (?<remote_p>.*?)::(?<remote_u>.*?)::(?<branch>.*)$",
                 RegexOptions.Compiled | RegexOptions.Multiline | RegexOptions.IgnorePatternWhitespace | RegexOptions.ExplicitCapture);
         private readonly string _refFormat = @"%(push:track,nobracket)::%(upstream:track,nobracket)::%(push)::%(upstream)::%(refname:short)";
+        private Lazy<IDictionary<string, AheadBehindData>?> _lazyData;
+        private string _branchName;
 
         public AheadBehindDataProvider(Func<IExecutable> getGitExecutable)
         {
             _getGitExecutable = getGitExecutable;
         }
 
+        public void InitLazy()
+        {
+            _lazyData = null;
+            _branchName = null;
+        }
+
         public IDictionary<string, AheadBehindData>? GetData(string branchName = "")
         {
             if (!AppSettings.ShowAheadBehindData)
             {
                 return null;
             }
 
-            return GetData(null, branchName);
+            // Callers setting branchname has the responisibility to ensure that not all are needed
+            if (string.IsNullOrWhiteSpace(branchName) && !string.IsNullOrWhiteSpace(_branchName))
+            {
+                // Debug.Fail($"Unexpectedly call for all branches after cache filled with specific branch {_branchName}");
```

Comment:
Is this needed?

Reply:
The commented out code is to document the behavor. I would have liked to enable the Debug if that had been working in .NET > 4.
For now it auto-resets the cache to get results for the branches.
Should just occur in tests.